src/types/utils.ts

```diff
@@ -0,0 +1 @@
+export type DeepWriteable<T> = { -readonly [P in keyof T]: DeepWriteable<T[P]> }
```

Comment:
Where can I read on the meaning of `-` here? `-readonly`?

Reply:
It's a "mapping modifier": https://www.typescriptlang.org/docs/handbook/2/mapped-types.html#mapping-modifiers